### PR TITLE
feat: Allow mouse to enter popover in charts

### PR DESCRIPTION
**Pull request by avinashbot**
### Description

The issue in question is that for screen magnification users, hovering over a point (or a vertical line) presents a popover, but when they move the mouse to read the popover, it goes away (because it's tracking only the point being hovered on). The solution for this is to have an area around the popover that disables the hover logic when the pointer is on top of it.

Related links, issue #, if available: AWSUI-19236, AWSUI-19205

### How has this been tested?

Added integ/unit tests and changed the behaviour for existing tests

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/area-chart/model/use-chart-model.ts [hunk 1]
```diff
@@ -18,6 +18,7 @@ import { throttle } from '../../internal/utils/throttle';
 
 const MAX_HOVER_MARGIN = 6;
 const SVG_HOVER_THROTTLE = 25;
+const POPOVER_DEADZONE = 12;
```
Comment (pan-kot) [inline src/area-chart/model/use-chart-model.ts:21]: Does it make sense to declare it somewhere under src/internal/components/chart-popover?

**Review by pan-kot — APPROVED** [review]
The code looks good to me, but I am not a fan of the resulting experience to be honest.

File: src/area-chart/__tests__/area-chart-use-chart-model.test.tsx [hunk 2]
```diff
@@ -14,9 +14,11 @@ import PlotPoint = ChartModel.PlotPoint;
 
 class UseChartModelWrapper extends ElementWrapper {
   static selectors = {
-    highlightedPoint: 'highlighted-point',
+    highlightedPoint: 'fhighlighted-point',
```
Comment (Al-Dani) [inline src/area-chart/__tests__/area-chart-use-chart-model.test.tsx:17]: is it a typo?

**Comment by abdhalees** [comment]
@pan-kot I added more unit tests to reach the threshold. no implementation changes

File: src/pie-chart/__tests__/pie-chart.test.tsx [hunk 3]
```diff
@@ -432,6 +433,74 @@ describe('Details popover', () => {
...
+    expect(detailPopover?.findHeader()?.getElement()).toHaveTextContent(defaultData[0].title);
+
+    act(() => {
+      fireEvent.keyDown(document, { key: 'Escape' });
+    });
+
+    expect(wrapper.findDetailPopover()).toBeNull();
+  });
+
+  test('pressing escape closes the popover', () => {
```
[the remark this answers is not in the document]
Comment (abdhalees) [reply src/pie-chart/__tests__/pie-chart.test.tsx]: copy pasting 🤦🏻‍♂️

**Pull request merged**